nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/PropertyCommentManager.java

```diff
@@ -132,8 +135,8 @@ public void deleteComment(DocumentModel docModel, DocumentModel comment) {
     public DocumentModel getThreadForComment(DocumentModel comment) {
         CoreSession session = comment.getCoreSession();
         DocumentModel parent = session.getDocument(new IdRef((String) comment.getPropertyValue(COMMENT_PARENT_ID)));
-        while (COMMENT_DOC_TYPE.equals(parent.getType())) {
-            comment = getThreadForComment(parent);
+        if (COMMENT_DOC_TYPE.equals(parent.getType())) {
+            return getThreadForComment(parent);
```

Comment:
Looks like you're fixing something that as wrong, could you explain why ?

Reply:
It was getting to an infinite cycle with scenarios of several nested replies.
I added a unit test that you can easily change and see that situation ;)